impeller/display_list/skia_conversions.h

```diff
@@ -46,5 +47,23 @@ Path PathDataFromTextBlob(const sk_sp<SkTextBlob>& blob);
 
 std::optional<impeller::PixelFormat> ToPixelFormat(SkColorType type);
 
+/// @brief Convert display list colors + stops into impeller colors and stops,
+/// taking care to ensure that the stops monotonically increase from 0.0 to 1.0.
+///
+/// The general process is:
+/// * Ensure that the first gradient stop value is 0.0. If not, insert a new
+///   stop with a value of 0.0 and use the first gradient color as this new
+///   stops color.
+/// * Ensure the last gradient stop value is 1.0. If not, insert a new stop
+///   with a value of 1.0 and use the last gradient color as this stops color.
+/// * Clamp all gradient values between the values of 0.0 and 1.0.
+/// * For all stop values, ensure that the values are monotonically increasing
+///   by clamping each value to a minimum of the previous stop value and itself.
+///   For example, with stop values of 0.0, 0.5, 0.4, 1.0, we would clamp such
```

Comment:
This bit is surprising to me. I would have expected it to be sorted but this works too.

Reply:
Still WIP and confirming :)

Reply:
What did you end up finding out here?

Reply:
I found the w3c notes which basically confirm this behavior is approximately correct